tests/by-util/test_skill.rs

```diff
@@ -0,0 +1,152 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+#[cfg(unix)]
+use crate::common::util::TestScenario;
+
+#[cfg(target_os = "linux")]
+#[test]
+fn test_missing_expression() {
+    new_ucmd!().fails().no_stdout().code_is(1).stderr_contains(
+        "the following required arguments were not provided:
+  <expression>...",
+    );
+}
+#[cfg(target_os = "linux")]
+#[test]
+fn test_default_signal() {
+    new_ucmd!()
+        .arg("-nv") // no action + verbose
+        .arg("1")
+        .succeeds()
+        .stdout_contains("Would send signal TERM to process 1");
+}
+
+#[cfg(target_os = "linux")]
+#[test]
+fn test_invalid_signal() {
+    new_ucmd!()
+        .arg("-INVALID")
+        .fails()
+        .code_is(2)
+        .no_stdout()
+        .stderr_contains("Invalid signal: INVALID");
+}
+#[cfg(target_os = "linux")]
+#[test]
+fn test_mutiple_signal() {
+    new_ucmd!()
+        .arg("-HUP")
+        .arg("-TERM")
+        .fails()
+        .code_is(2)
+        .no_stdout()
+        .stderr_contains("Too many signals");
+}
+#[cfg(target_os = "linux")]
+#[test]
+fn test_verbose_option() {
+    new_ucmd!()
+        .arg("-n") // no action
+        .arg("-v")
+        .arg("1")
+        .succeeds()
+        .stdout_contains("Would send signal TERM to process 1")
+        .no_stderr();
+}
+#[cfg(target_os = "linux")]
+#[test]
+fn test_list_option() {
+    new_ucmd!()
+        .arg("-l")
+        .succeeds()
+        .no_stderr()
+        .stdout_contains("HUP INT QUIT ILL TRAP ABRT BUS FPE KILL USR1 SEGV USR2 PIPE ALRM TERM STKFLT CHLD CONT STOP TSTP TTIN TTOU URG XCPU XFSZ VTALRM PROF WINCH POLL PWR SYS");
+}
+
+#[cfg(target_os = "linux")]
+#[test]
+fn test_list_option_long() {
+    new_ucmd!()
+        .arg("--list")
+        .succeeds()
+        .no_stderr()
+        .stdout_contains("HUP INT QUIT ILL TRAP ABRT BUS FPE KILL USR1 SEGV USR2 PIPE ALRM TERM STKFLT CHLD CONT STOP TSTP TTIN TTOU URG XCPU XFSZ VTALRM PROF WINCH POLL PWR SYS");
+}
```

Comment:
Here it might make sense to combine the functions into one and simply loop through the args, something like:
```rust
for arg in ["-l", "--list"] {
    new_ucmd!()
        .arg(arg)
...
}
```